### PR TITLE
OHRM-240:Google OpenID support bug fixed

### PR DESCRIPTION
**Pull request by supunrandunurk**

### CODE REVIEW

File: symfony/plugins/orangehrmOpenidAuthenticationPlugin/lib/authproviders/GoogleAuthProvider.php [hunk 1]
```diff
@@ -23,43 +23,40 @@
...
         $gClient->setRedirectUri($provider->getProviderUrl());
         $gClient->setDeveloperKey($authProvider->getDeveloperKey());
-        $gClient->addScope(array(self::EMAIL_SCOPE,self::PROFILE_SCOPE));
-
-        $gService = new Google_Service_Plus($gClient);
+        $gClient->addScope(array(self::EMAIL_SCOPE, self::PROFILE_SCOPE));
 
         if (isset($_GET['code'])) {
-            $gClient->authenticate($_GET['code']);
+            $gClient->fetchAccessTokenWithAuthCode($_GET['code']);
```
Comment (samanthajayasinghe) [inline symfony/plugins/orangehrmOpenidAuthenticationPlugin/lib/authproviders/GoogleAuthProvider.php]: It's not good to couple GET params inside the method.
better to pass it as parameter or setter method for the class

File: symfony/plugins/orangehrmOpenidAuthenticationPlugin/lib/authproviders/GoogleAuthProvider.php [hunk 2]
```diff
@@ -23,43 +23,40 @@
  * Description of GoogleAuthProvider
  */
 class GoogleAuthProvider extends AbstractAuthProvider {
-
     const PROFILE_SCOPE = 'profile';
     const EMAIL_SCOPE = 'email';
+    protected $loginService;
```
Comment (samanthajayasinghe) [inline symfony/plugins/orangehrmOpenidAuthenticationPlugin/lib/authproviders/GoogleAuthProvider.php]: the default value is missing 
protected $loginService = null

File: symfony/plugins/orangehrmOpenidAuthenticationPlugin/lib/authproviders/GoogleAuthProvider.php [hunk 3]
```diff
@@ -23,43 +23,40 @@
...
-            $dataArray['useridentity'] = $gClient->getAccessToken();
+            $dataArray['useridentity'] = json_encode($gClient->getAccessToken());
             $success = $this->getOpenIdService()->setOpenIdCredentials($username, $dataArray);
             if ($success) {
+                $this->getLoginService()->addLogin();
                 $flag = array('type' => 'true', 'message' => 'User has authentication!');
                 return $flag;
             } else {
-                $flag = array('type' => 'false', 'message' => 'User Account Not found');
+                $flag = array('type' => 'false', 'message' => 'Invalid Credentials : you Have No OpenID account in Orangehrm try loging with OrangeHRM credentials');
```
Comment (samanthajayasinghe) [inline symfony/plugins/orangehrmOpenidAuthenticationPlugin/lib/authproviders/GoogleAuthProvider.php]: better to add a multiline array for readability

**Comment by icechathuranga** [comment]
need to do code refactoring

**Pull request closed without merging**